### PR TITLE
Unify corerun's implementation of the native debugger check with the minipal for all platforms

### PR DESCRIPTION
**Pull request by jkoritzinsky**
Deduplicate the non-Windows path so new platforms only need to add support in one place.

### CODE REVIEW

File: src/native/minipal/debugger.c [hunk 1]
```diff
@@ -17,28 +17,43 @@
...
 #include <OS.h>
+#define MINIPAL_DEBUGGER_PRESENT_CHECK
 #endif
 
+bool minipal_can_check_for_native_debugger(void)
+{
+#if defined(MINIPAL_DEBUGGER_PRESENT_CHECK)
+    return true;
+#else
+    return false;
```
Comment (am11) [inline src/native/minipal/debugger.c:54]:
```suggestion
    return false;
#endif
```

File: src/native/minipal/debugger.c [hunk 2]
```diff
@@ -17,28 +17,43 @@
...
 #include <sys/procfs.h>
+#define MINIPAL_DEBUGGER_PRESENT_CHECK
 #elif defined(__HAIKU__)
 #include <OS.h>
+#define MINIPAL_DEBUGGER_PRESENT_CHECK
 #endif
 
+bool minipal_can_check_for_native_debugger(void)
+{
+#if defined(MINIPAL_DEBUGGER_PRESENT_CHECK)
```
Comment (am11) [inline src/native/minipal/debugger.c:51]: Looks like an always-true condition?

Comment (jkoritzinsky) [reply src/native/minipal/debugger.c:51]: It’s true for every platform we support, but if you build on an unsupported platform, we’ll hit the false case.

**Review by AaronRobinsonMSFT — APPROVED** [review]

File: src/native/minipal/debugger.c (hunk 2 above)
Comment (am11) [reply src/native/minipal/debugger.c:51]: IMO, we should just delete `minipal_can_check_for_native_debugger` and add `#else\n#error "Not yet implemented"` on line 47 to let the person/team consciously make a determination during the porting time (whether they way to ignore this feature by adding `#elif defined(__someOS__) /* ignored */` or implement it as part of the port).

Comment (am11) [reply src/native/minipal/debugger.c:51]: IMO, just delete `minipal_can_check_for_native_debugger` and add `#else\n#error "Not yet implemented"` on line 47 to let the person/team consciously make a determination during the porting time (whether they want to ignore this feature by adding `#elif defined(__someOS__) /* ignored */` or implement it as part of the port).

**Comment by jkoritzinsky** [comment]
/ba-g debugger helix workitem deadletter (which we can't use known issues for today)

File: src/native/minipal/debugger.c (hunk 2 above)
Comment (am11) [reply src/native/minipal/debugger.c:51]: @AaronRobinsonMSFT, without the additional API: https://github.com/dotnet/runtime/compare/main...am11:runtime:dedup-debugger-check, it wouldn't block HelloWorld during the port to new platform. To get by, the person will add an empty elif branch likely with a TODO (as they would need to adjust tons of other places) and obviously not use `-d` or `--debug` options with `corerun`, as `wait_for_debugger` is only called in that case. 

No big deal, but that build-time error is the more prevalent pattern used in coreclr (I remembered hitting them in past adventures 😅).

**Pull request merged**